test/storage/sql_table_test.cpp

```diff
@@ -489,6 +506,7 @@ TEST_F(SqlTableTests, UpdateTest) {
 TEST_F(SqlTableTests, ScanTest) {
   std::map<uint32_t, uint32_t> datname_map;
   std::map<uint32_t, uint32_t> new_col_map;
+  uint32_t new_col_default_value = 1729;
```

Comment:
I think we should draw default values out into a separate test suite.  There are several edge cases and sequencing concerns that I think we need to explicitly test.  Specifically, things like: (1) column added without a default value has one in a later schema change (we should ignore this since it is not retroactive), (2) column added with a default value does not have one in a later schema change (we need to maintain the old default and continue to apply it to the tables that don't have the column), (3) VARLEN default values for an added column that cannot be inlined (i.e. more than 12 characters long) [has possible reference counting/correctness concerns], and possibly others.

Reply:
Yeah, you are right. I'll make more comprehensive test cases for the default values. I have a few questions about the points you made though:
1) How is this change of default value going to be implemented? Should I be exposing a `setDefaultValue` method? In that case, I have to find some other flag to indicate whether the default value in the `Column` is assigned at the time of creation or later.
2) Again, should I expose an API like `clearDefaultValue`? But this shouldn't actually clear the default values from the `DefaultValueMap` in the `DataTableVersion`.
3) I'll check the code about how VARLEN reference counting is handled. It'll be a very bad idea to copy the VARLEN content as I'm doing it now.